src/core/dbus-cgroup.c

```diff
@@ -1312,11 +1312,9 @@ int bus_cgroup_set_property(
                         if (c->cpu_quota_per_sec_usec == USEC_INFINITY)
                                 unit_write_setting(u, flags, "CPUQuota", "CPUQuota=");
                         else
-                                /* config_parse_cpu_quota() requires an integer, so truncating division is used on
-                                 * purpose here. */
                                 unit_write_settingf(u, flags, "CPUQuota",
-                                                    "CPUQuota=%0.f%%",
-                                                    (double) (c->cpu_quota_per_sec_usec / 10000));
+                                                    "CPUQuota=%.2f%%",
```

Comment:
hmm, to avoid l10n issues can you please output this as fixed point, i.e.:

```c
CPUQuota=$f%.%02u
```

and then calculate the rhs appropriately via modulo?

Reply:
(background: %f uses "," or "." as separator for the fractional part, depending on l10n

Reply:
@poettering Your suggestion does not work and I can't find any documentation on this syntax so I'm not really sure what it should be. I get a warning now saying the data argument is not used by the format string.

Reply:
```c
unit_write_setting(…, "CPUQuota=%0.f.%02" PRIu64 "%%, (double) c->cpu_quota_per_sec_usec / 10000, ((uint64_t) c->cpu_quota_per_sec_usec / 100U) % 100U);
```

Reply:
@poettering I came up with something slightly different, PTAL!